client/app/services/dashboard.js

```diff
@@ -171,6 +172,7 @@ const DashboardService = {
   favorites: params => axios.get("api/dashboards/favorites", { params }).then(transformResponse),
   favorite: ({ id }) => axios.post(`api/dashboards/${id}/favorite`),
   unfavorite: ({ id }) => axios.delete(`api/dashboards/${id}/favorite`),
+  recentDashboards: params => axios.get("api/dashboards", { params }).then(transformResponse).then((data) => filterRecent(data, "dashboard"))
```

Comment:
As far as I understand - it currently doesn't work with pagination, right?

Reply:
It currently doesn't work with pagination, but i don't think that we will need an pagination for it too, the objects are limited for 5 of each type like query or dashboard so the list will have like 20 items max. The big problem is that we need an endpoint to retrieve the recent items, the way this is currently working we receive the response of get api/dashboards and filter this list searching for items that are in the "recents" array, but with the pagination, we can have items that are in the recents array but aren't being retrieved in the first page of the request response.. I was thinking that implementing an endpoint that receive an array of ids and an object type to retrieve filtered items could be an good alternative or maybe an endpoint that retrieve all user querys, dashboards and alerts so we can filter this on frontend if we're following with the unified list idea.

Reply:
Filtering it on frontend is definitely not a good idea as there may be really a lot of objects, and loading all of them just to pick five of them is totally a overkill. In this case it's even better to load objects individually by IDs (making requests in parallel of course) - it will be less load for a backend than getting all. But having a dedicated API endpoint feels like the best idea in this case. You can try to implement it - it's not that hard, but also feel free to ask for a help if needed.

Reply:
For sure this would be an overkill hahah, i will try to implement this endpoint!

Reply:
Hi @kravets-levko, I just finished the implementations that you suggested about the "addToRecentsAtLocalStorage" being a class! This week i will try to work on the endpoint.